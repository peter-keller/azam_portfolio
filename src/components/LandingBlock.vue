<template>
  <div class='landing-block' :style="`background-image: url(${image})`">
    <div class="headline">
      <h1>{{ headline }}</h1>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    image: {
      type: String,
      required: true,
      default: 'https://wallpaperaccess.com/full/405435.jpg'
    },

    headline: {
      type: String,
      required: false,
      default: 'I am Azam Khan'
    }
  }
}
</script>

<style lang="scss" scoped>

  .landing-block {
    width: 100vw;
    height: 100vh;
    background-repeat: no-repeat;
    background-attachment: fixed;
    background-size: cover;
    background-position: center center;
    display: flex;
    justify-content: center;
    align-items: center;

    h1 {
      font-size: 64px;
      margin: 0;
      color: $color-white;
    }
  }

</style>
