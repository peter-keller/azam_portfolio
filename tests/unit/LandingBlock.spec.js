import { shallowMount } from '@vue/test-utils'
import LandingBlock from '../../src/components/LandingBlock'


describe('/components/landing-block', () => {
  const factory = (opts = {}) => shallowMount(LandingBlock, {
    ...opts
  })

  it('matches snapshot', () => {
    const wrapper = factory()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches snapshot with custom props', () => {
    const wrapper = factory({
      propsData: {
        headline: 'Test headline',
        image: 'image src'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
