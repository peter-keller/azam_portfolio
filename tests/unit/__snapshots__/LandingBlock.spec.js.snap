// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/landing-block matches snapshot 1`] = `
<div class="landing-block" style="background-image: url(https://wallpaperaccess.com/full/405435.jpg);">
  <div class="headline">
    <h1>I am Azam Khan</h1>
  </div>
</div>
`;

exports[`/components/landing-block matches snapshot with custom props 1`] = `
<div class="landing-block">
  <div class="headline">
    <h1>Test headline</h1>
  </div>
</div>
`;
